src/lib/operators/abstract_operator.hpp

```diff
@@ -118,6 +118,10 @@ class AbstractOperator : public std::enable_shared_from_this<AbstractOperator>,
   std::shared_ptr<const Table> input_table_left() const;
   std::shared_ptr<const Table> input_table_right() const;
 
+  void set_lqp_node(const std::shared_ptr<const AbstractLQPNode> lqp_node);
+
+  std::shared_ptr<const AbstractLQPNode> get_lqp_node() const;
```

Comment:
Any reason we need setters and getters?

Reply:
For the most part, we have switched to directly modifying the field: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rh-get

Reply:
Done.